packages/components/src/unit-control/utils.js

```diff
@@ -9,6 +9,17 @@ import { isEmpty } from 'lodash';
 import { __, _x } from '@wordpress/i18n';
 import { Platform } from '@wordpress/element';
 
+/**
+ * An object containing the details of a unit.
+ *
+ * @typedef {Object} WPUnitControlUnit
+ * @property {string}        value       The value for the unit, used in a CSS value (e.g `px`).
+ * @property {string}        label       The label used in a dropdown selector for the unit.
+ * @property {string|number} [default]   Default value for the unit, used when switching units.
+ * @property {string}        [a11yLabel] An accessible label used by screen readers.
+ * @property {number}        [step]      A step value used when incrementing/decrementing the value.
+ */
```

Comment:
Great job! Just one curiosity — how did you determine that `default`, `a11yLabel` and `step` are optional properties?

Also, could you please also update [the `units` prop docs in the README](https://github.com/WordPress/gutenberg/blob/b61a2d845fbb3e1434c9bd2f611293e8e3cced09/packages/components/src/unit-control/README.md?plain=1#L105)? I guess `Array<WPUnitControlUnit>` would already be a better option

Reply:
Thanks for catching that, I've updated the README. And good questions, I determined the required vs optional properties in the following way from looking at `index.js` and `index.native.js`, but let me know if I've missed something and I'm happy to update it!

* The `default` prop is optional because [on this line](https://github.com/WordPress/gutenberg/blob/14d80aa66819682118a73f3cb40485806b9b8c44/packages/components/src/unit-control/index.js#L76) it is only used when switching units if it's set
* The `a11yLabel` prop is optional because it isn't used in the web version (so isn't currently required there), and in the only place it's used in Native [it falls back](https://github.com/wordpress/gutenberg/blob/e9e62d3a10d57b581ca88ca89ee46a88d5d8507d/packages/components/src/unit-control/utils.js#L238) to the regular label if it doesn't exist
* The `step` prop is optional because [on this line](https://github.com/wordpress/gutenberg/blob/14d80aa66819682118a73f3cb40485806b9b8c44/packages/components/src/unit-control/index.js#L165) the default step is set if one doesn't exist.

Reply:
Thank you for the explanation, it makes sense!

> The default prop is optional because on this line it is only used when switching units if it's set

I've also noticed this sentence in the README which confirms your point: "A `default` value (in the example above, `10` for `%`), **if defined**, is set as the new `value` when a unit changes.`